src/vs/workbench/parts/terminal/electron-browser/terminalConfigHelper.ts

```diff
@@ -67,34 +69,73 @@ const DEFAULT_ANSI_COLORS = {
 	]
 };
 
+export interface ITerminalFont {
+	fontFamily: string;
+	fontSize: number;
+	lineHeight: number;
+	charWidth: number;
+	charHeight: number;
+}
+
 /**
  * Encapsulates terminal configuration logic, the primary purpose of this file is so that platform
  * specific test cases can be written.
  */
 export class TerminalConfigHelper {
+	private charMeasureElement: HTMLElement;
+
 	public constructor(
 		private platform: Platform,
 		private configurationService: IConfigurationService,
-		private themeService: IThemeService) {
+		private themeService: IThemeService,
+		private parentDomElement: HTMLElement) {
 	}
 
 	public getTheme(): string[] {
 		let baseThemeId = getBaseThemeId(this.themeService.getTheme());
 		return DEFAULT_ANSI_COLORS[baseThemeId];
 	}
 
+	private neasureFont(fontFamily: string, fontSize: number, lineHeight: number): ITerminalFont {
+		if (!this.charMeasureElement) {
+			this.charMeasureElement = new Builder(this.parentDomElement, true).div().build().getHTMLElement();
+		}
+		let style = this.charMeasureElement.style;
+		style.display = 'inline';
+		style.fontFamily = fontFamily;
+		style.fontSize = fontSize + 'px';
+		style.lineHeight = lineHeight + 'px';
+		this.charMeasureElement.innerText = 'X';
+		let rect = this.charMeasureElement.getBoundingClientRect();
+		style.display = 'none';
+		let charWidth = Math.ceil(rect.width);
+		let charHeight = Math.ceil(rect.height);
+		return {
+			fontFamily,
+			fontSize,
+			lineHeight,
+			charWidth,
+			charHeight
+		};
+	}
+
 	/**
-	 * Set the terminal font to `terminal.integrated.fontFamily` if it is set, otherwise fallback to
-	 * `editor.fontFamily`.
+	 * Gets the font information based on the terminal.integrated.fontFamily,
+	 * terminal.integrated.fontSize, terminal.integrated.lineHeight configuration properties
 	 */
-	public getFontFamily(): string {
-		let terminalConfig = this.configurationService.getConfiguration<ITerminalConfiguration>();
-		let fontFamily = terminalConfig.terminal.integrated.fontFamily;
-		if (!fontFamily) {
-			let editorConfig = this.configurationService.getConfiguration<IConfiguration>();
-			fontFamily = editorConfig.editor.fontFamily;
+	public getFont(): ITerminalFont {
+		let terminalConfig = this.configurationService.getConfiguration<ITerminalConfiguration>().terminal.integrated;
+		let editorConfig = this.configurationService.getConfiguration<IConfiguration>();
+
+		let fontFamily = terminalConfig.fontFamily || editorConfig.editor.fontFamily;
+		let fontSize = this.toInteger(terminalConfig.fontSize, 0) || editorConfig.editor.fontSize;
+		let lineHeight = this.toInteger(terminalConfig.lineHeight, 0) || editorConfig.editor.lineHeight;
+
+		if (lineHeight === 0) {
+			lineHeight = Math.round(GOLDEN_LINE_HEIGHT_RATIO * fontSize);
```

Comment:
We basically want consistency with how editor.lineHeight works here, is this how that falls back?